#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.0054464034566594835 0.6937620828834817 0.7201836634099391 3.1226430700959082
  position 0.29186455791363614 5.413845196108004 0.24912669078852945
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 3.5 3.5
  wallThickness 0.02
  wallHeight 0.3
}
Robot {
  translation -1.5567102755715272 0.0397206085414068 1.1885859808949575
  rotation 2.6240334327930767e-05 0.9999999994669775 -1.9429102022065136e-05 -1.8657490967037134
  children [
    GPS {
    }
    Camera {
      translation 0 0.03 0.085
      rotation 0 1 0 3.14159
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
          }
          geometry Box {
            size 0.04 0.01 0.01
          }
        }
      ]
      name "camera0"
      width 100
      height 100
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.678431 0.498039 0.658824
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        position 179.2639344653845
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.059999999603073784 -2.854417133032927e-06 0.05000269263808757
        rotation -0.7059329956090291 0.7059355887073495 0.05756344592595719 3.0265930907113714
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0.203922 0.396078 0.643137
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 4.490748906023698e-09 3.6202134276098825e-08 1.2902879718109523e-09
        angularVelocity 3.224488986149608e-08 2.1557990475276574e-11 -1.1232651346559002e-07
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        position 170.88006209110733
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.0600000003974106 3.595854379952917e-06 0.04999842920827169
        rotation -0.4933293044325204 0.493331116630753 -0.7164151078473847 4.384905315914677
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 4.488611983156752e-09 3.596203165875569e-08 1.2895338623021274e-09
        angularVelocity 3.2214602194497346e-08 2.1563773983453872e-11 -1.1222658170653848e-07
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        position 179.26393786552165
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.060000000397222086 -2.85438840368695e-06 -0.049997307331450705
        rotation -0.7059329460986922 0.7059355392145887 0.057564660046148304 3.026590670473263
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 4.493642702377996e-09 2.452151568559684e-08 1.288884286128835e-09
        angularVelocity 3.224495533626633e-08 2.155764578776308e-11 -1.1232669183049241e-07
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        position 170.8800654398489
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.05999999960326252 3.595837682635856e-06 -0.050001570829962615
        rotation -0.49333021970639257 0.49333203188822766 -0.7164138473239299 4.3849036499099405
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
        linearVelocity 4.491504855586252e-09 2.428141900528507e-08 1.288129457634648e-09
        angularVelocity 3.22146657067792e-08 2.1564103828453796e-11 -1.1222675817760686e-07
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.03 0 0.1
      rotation 0 1 0 -1.87079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.988235 0.913725 0.309804
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.03 0 0.1
      rotation 0 1 0 -1.27079
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.988235 0.913725 0.309804
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "collision_avoidance"
}
